accepted/0000-link-packages-to-source-and-build.md

```diff
@@ -0,0 +1,528 @@
+
+The signing and verification logic will be implemented in a [sigstore-js](https://github.com/sigstore/sigstore-js) client library using Node.js's crypto library. Similar clients exist in [Go](https://github.com/sigstore/cosign), [Python](https://github.com/sigstore/sigstore-python) and [Java](http://github.com/sigstore/sigstore-java) and we plan to follow their lead on implementation details.
+
+The npm CLI will rely on the Sigstore JS client library to generate the provenance attestation on publish and to verify the detached signature bundle that's downloaded from the registry.
+
+The Sigstore client library will include direct vendor support for requesting an OIDC id token from supported CI/CD systems, similar to the existing client libraries. It will also include all the required functionality to interact with Fulcio and Rekor, as well as verify the [Sigstore root keys](https://github.com/sigstore/root-signing) using a [TUF](https://theupdateframework.github.io/) client.
+
+#### Fetching Sigstore hosts from the configured npm registry
+The npm CLI will fetch Sigstore config from the configured npm registry that contains the public Fulcio and Rekor hosts by default:
+
+```
+GET https://registry-host.tld/-/npm/v1/sigstore-config
+
+{
+  "sigstoreConfig": {
+    "fulcioUrl": "https://fulcio.sigstore.dev",
+    "rekorUrl": "https://rekor.sigstore.dev",
+    "tufRepository": "https://sigstore-tuf-root.storage.googleapis.com",
+    "tufRoot": "root.json"
+  }
+}
+```
+
+This would allow future extensions where maintainers can configure the instance being used, making sure the release attestation gets placed on the same Rekor instance as the build attestation. The release attestation will be generated by the registry when the package has been successfully authorized and published.
+
+#### Publishing
+Maintainers will need to enable build provenance generation with a command line flag: `npm publish --with=build-signatures` in addition to running it in a supported CI/CD system. The command will error with an error message if this is not run in a supported build system.
+
+Once enabled, the CLI will perform the required steps to generate the provenance attestation and upload this to the public Rekor ledger for transparency and a detached copy that contains proof of inclusion from Rekor will be stored in the npm registry.
+
+##### What goes on the public Rekor ledger?
+Only public npm packages will be signed and published to the public [rekor.sigstore.dev](https://docs.sigstore.dev/rekor/overview) ledger by default.
+
+Privately scoped packages, or packages from private repositories, will not be signed or published to the public ledger. This will be determined by interrogating the npm registry at publish time. Users will be able to override this similar to the `cosign --force` command by passing an argument to `npm publish`, e.g. `--force-build-signatures`.
```

Comment:
Why is `--force-build-signatures` required when `--with=build-signatures` is not enabled by default?

Reply:
> Why is `--force-build-signatures` required when `--with=build-signatures` is not enabled by default?

This was meant for private packages, that by default don't upload provenance information on the immutable public ledger Rekor, even if you say `--with=build-signatures`. Not sure we need this unless we have users who actually want to upload this information about private packages.

Reply:
Would it be possible to use something a bit more descriptive, like `--force-publish-signatures` instead? It would eliminate confusion.